### PR TITLE
Remove WorldManager

### PR DESCRIPTION
**Pull request by tinchou**
This removes the unnecessary `WorldManager` interface and `WorldManagerImpl` class from production code. It was used just as an object holder, hiding inter-component dependencies that are now explicit. The `SimpleWorldManager` class is left as a test helper.

As a result, some constructors are growing even more, but this should be tackled as a separate task: refactor components into a more cohesive architecture, where each component doesn't need tens of collaborators.

### CODE REVIEW

**Comment by rskops** [comment]
SonarQube analysis reported 171 issues
* ![MAJOR][MAJOR] 129 major
* ![MINOR][MINOR] 21 minor
* ![INFO][INFO] 21 info

#### Top 10 issues

1. ![MAJOR][MAJOR] [Start.java#L89](https://github.com/rsksmart/rskj/blob/d44e0c467ec584486c1ac6c429a10b65e963afeb/rskj-core/src/main/java/co/rsk/Start.java#L89): Constructor has 15 parameters, which is greater than 7 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [Start.java#L121](https://github.com/rsksmart/rskj/blob/d44e0c467ec584486c1ac6c429a10b65e963afeb/rskj-core/src/main/java/co/rsk/Start.java#L121): The Cyclomatic Complexity of this method "startNode" is 12 which is greater than 10 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [Start.java#L121](https://github.com/rsksmart/rskj/blob/d44e0c467ec584486c1ac6c429a10b65e963afeb/rskj-core/src/main/java/co/rsk/Start.java#L121): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)
1. ![MAJOR][MAJOR] [RskFactory.java#L111](https://github.com/rsksmart/rskj/blob/d44e0c467ec584486c1ac6c429a10b65e963afeb/rskj-core/src/main/java/co/rsk/core/RskFactory.java#L111): This block of commented-out lines of code should be removed. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
1. ![MAJOR][MAJOR] [RskFactory.java#L134](https://github.com/rsksmart/rskj/blob/d44e0c467ec584486c1ac6c429a10b65e963afeb/rskj-core/src/main/java/co/rsk/core/RskFactory.java#L134): Method has 17 parameters, which is greater than 7 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [RskImpl.java#L43](https://github.com/rsksmart/rskj/blob/d44e0c467ec584486c1ac6c429a10b65e963afeb/rskj-core/src/main/java/co/rsk/core/RskImpl.java#L43): Constructor has 11 parameters, which is greater than 7 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [BlockChainImpl.java#L100](https://github.com/rsksmart/rskj/blob/d44e0c467ec584486c1ac6c429a10b65e963afeb/rskj-core/src/main/java/co/rsk/core/bc/BlockChainImpl.java#L100): Constructor has 8 parameters, which is greater than 7 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [BlockChainImpl.java#L197](https://github.com/rsksmart/rskj/blob/d44e0c467ec584486c1ac6c429a10b65e963afeb/rskj-core/src/main/java/co/rsk/core/bc/BlockChainImpl.java#L197): The Cyclomatic Complexity of this method "internalTryToConnect" is 22 which is greater than 10 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [BlockChainImpl.java#L199](https://github.com/rsksmart/rskj/blob/d44e0c467ec584486c1ac6c429a10b65e963afeb/rskj-core/src/main/java/co/rsk/core/bc/BlockChainImpl.java#L199): Method co.rsk.core.bc.BlockChainImpl.internalTryToConnect(Block) appears to call the same method on the same object redundantly [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. ![MAJOR][MAJOR] [BlockChainImpl.java#L311](https://github.com/rsksmart/rskj/blob/d44e0c467ec584486c1ac6c429a10b65e963afeb/rskj-core/src/main/java/co/rsk/core/bc/BlockChainImpl.java#L311): Method co.rsk.core.bc.BlockChainImpl.internalTryToConnect(Block) appears to call the same method on the same object redundantly [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: rskj-core/src/test/java/co/rsk/core/BlockchainDummy.java [hunk 1]
```diff
@@ -55,6 +56,11 @@ public Block getBestBlock() {
         return null;
     }
 
+    @Override
+    public void setBlockRecorder(BlockRecorder blockRecorder) {
+
```
Comment (diega) [inline rskj-core/src/test/java/co/rsk/core/BlockchainDummy.java]: what if we throw `java.lang.UnsupportedOperationException` here?

Comment (tinchou) [reply rskj-core/src/test/java/co/rsk/core/BlockchainDummy.java]: This is test code, but I'm fine either way.

File: rskj-core/src/test/java/org/ethereum/rpc/Simples/SimpleWorldManager.java [hunk 2]
```diff
@@ -21,17 +21,17 @@
...
 import org.ethereum.listener.CompositeEthereumListener;
 import org.ethereum.listener.EthereumListener;
-import org.ethereum.manager.WorldManager;
 import org.ethereum.net.client.ConfigCapabilities;
 
 /**
  * Created by Ruben Altman on 09/06/2016.
  */
-public class SimpleWorldManager implements WorldManager {
+public class SimpleWorldManager {
```
Comment (lsebrie) [inline rskj-core/src/test/java/org/ethereum/rpc/Simples/SimpleWorldManager.java]: Removing SimpleWorldManager should be much easier now, it should be added to this PR

**Comment by SergioDemianLerner** [comment]
Disclaimer: I have not checked the code of this change. 

But I have tried several times rebase code that adds a new object to the system state, and it requires filling 100 constructor arguments with one more null, making future rebase the most sad experience ever. We have so many constructors with docents of unnamed arguments!. 
Simple modifications to the objects in the state end up touching a large number of files (like 40). That is what WorldState was created for: to make simple changes affect very few parts of the code. Yes, it also has drawbacks, the "bag of cats" argument. But please consider those large, impossible to read, and easy to confuse constructors. Please consider the poor, sad and lonely programmer that has to modify all those constructor arguments.
I would do the exact opposite: put all objects in a WorldState "bag of cats", and leave them with null by default. Let anyone use the object it wants, and let every component check if the objects it uses are non-null at construction time.
This is what I would do, if I were an RSK Core programmer.

**Comment by rskops** [comment]
SonarQube analysis reported 171 issues
* ![MAJOR][MAJOR] 129 major
* ![MINOR][MINOR] 21 minor
* ![INFO][INFO] 21 info

#### Top 10 issues

1. ![MAJOR][MAJOR] [Start.java#L89](https://github.com/rsksmart/rskj/blob/a1cb4990b1367b01f8486a6523dd11f931391d3e/rskj-core/src/main/java/co/rsk/Start.java#L89): Constructor has 15 parameters, which is greater than 7 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [Start.java#L121](https://github.com/rsksmart/rskj/blob/a1cb4990b1367b01f8486a6523dd11f931391d3e/rskj-core/src/main/java/co/rsk/Start.java#L121): The Cyclomatic Complexity of this method "startNode" is 12 which is greater than 10 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [Start.java#L121](https://github.com/rsksmart/rskj/blob/a1cb4990b1367b01f8486a6523dd11f931391d3e/rskj-core/src/main/java/co/rsk/Start.java#L121): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)
1. ![MAJOR][MAJOR] [RskFactory.java#L111](https://github.com/rsksmart/rskj/blob/a1cb4990b1367b01f8486a6523dd11f931391d3e/rskj-core/src/main/java/co/rsk/core/RskFactory.java#L111): This block of commented-out lines of code should be removed. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
1. ![MAJOR][MAJOR] [RskFactory.java#L134](https://github.com/rsksmart/rskj/blob/a1cb4990b1367b01f8486a6523dd11f931391d3e/rskj-core/src/main/java/co/rsk/core/RskFactory.java#L134): Method has 17 parameters, which is greater than 7 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [RskImpl.java#L43](https://github.com/rsksmart/rskj/blob/a1cb4990b1367b01f8486a6523dd11f931391d3e/rskj-core/src/main/java/co/rsk/core/RskImpl.java#L43): Constructor has 11 parameters, which is greater than 7 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [BlockChainImpl.java#L100](https://github.com/rsksmart/rskj/blob/a1cb4990b1367b01f8486a6523dd11f931391d3e/rskj-core/src/main/java/co/rsk/core/bc/BlockChainImpl.java#L100): Constructor has 8 parameters, which is greater than 7 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [BlockChainImpl.java#L197](https://github.com/rsksmart/rskj/blob/a1cb4990b1367b01f8486a6523dd11f931391d3e/rskj-core/src/main/java/co/rsk/core/bc/BlockChainImpl.java#L197): The Cyclomatic Complexity of this method "internalTryToConnect" is 22 which is greater than 10 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [BlockChainImpl.java#L199](https://github.com/rsksmart/rskj/blob/a1cb4990b1367b01f8486a6523dd11f931391d3e/rskj-core/src/main/java/co/rsk/core/bc/BlockChainImpl.java#L199): Method co.rsk.core.bc.BlockChainImpl.internalTryToConnect(Block) appears to call the same method on the same object redundantly [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. ![MAJOR][MAJOR] [BlockChainImpl.java#L311](https://github.com/rsksmart/rskj/blob/a1cb4990b1367b01f8486a6523dd11f931391d3e/rskj-core/src/main/java/co/rsk/core/bc/BlockChainImpl.java#L311): Method co.rsk.core.bc.BlockChainImpl.internalTryToConnect(Block) appears to call the same method on the same object redundantly [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by rskops** [comment]
SonarQube analysis reported 171 issues
* ![MAJOR][MAJOR] 129 major
* ![MINOR][MINOR] 21 minor
* ![INFO][INFO] 21 info

#### Top 10 issues

1. ![MAJOR][MAJOR] [Start.java#L89](https://github.com/rsksmart/rskj/blob/892c99e1f7fc398495823679378f0f9f0bc504db/rskj-core/src/main/java/co/rsk/Start.java#L89): Constructor has 15 parameters, which is greater than 7 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [Start.java#L121](https://github.com/rsksmart/rskj/blob/892c99e1f7fc398495823679378f0f9f0bc504db/rskj-core/src/main/java/co/rsk/Start.java#L121): The Cyclomatic Complexity of this method "startNode" is 12 which is greater than 10 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [Start.java#L121](https://github.com/rsksmart/rskj/blob/892c99e1f7fc398495823679378f0f9f0bc504db/rskj-core/src/main/java/co/rsk/Start.java#L121): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)
1. ![MAJOR][MAJOR] [RskFactory.java#L112](https://github.com/rsksmart/rskj/blob/892c99e1f7fc398495823679378f0f9f0bc504db/rskj-core/src/main/java/co/rsk/core/RskFactory.java#L112): This block of commented-out lines of code should be removed. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
1. ![MAJOR][MAJOR] [RskFactory.java#L139](https://github.com/rsksmart/rskj/blob/892c99e1f7fc398495823679378f0f9f0bc504db/rskj-core/src/main/java/co/rsk/core/RskFactory.java#L139): Method has 17 parameters, which is greater than 7 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [RskImpl.java#L43](https://github.com/rsksmart/rskj/blob/892c99e1f7fc398495823679378f0f9f0bc504db/rskj-core/src/main/java/co/rsk/core/RskImpl.java#L43): Constructor has 11 parameters, which is greater than 7 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [BlockChainImpl.java#L100](https://github.com/rsksmart/rskj/blob/892c99e1f7fc398495823679378f0f9f0bc504db/rskj-core/src/main/java/co/rsk/core/bc/BlockChainImpl.java#L100): Constructor has 8 parameters, which is greater than 7 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [BlockChainImpl.java#L197](https://github.com/rsksmart/rskj/blob/892c99e1f7fc398495823679378f0f9f0bc504db/rskj-core/src/main/java/co/rsk/core/bc/BlockChainImpl.java#L197): The Cyclomatic Complexity of this method "internalTryToConnect" is 22 which is greater than 10 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [BlockChainImpl.java#L199](https://github.com/rsksmart/rskj/blob/892c99e1f7fc398495823679378f0f9f0bc504db/rskj-core/src/main/java/co/rsk/core/bc/BlockChainImpl.java#L199): Method co.rsk.core.bc.BlockChainImpl.internalTryToConnect(Block) appears to call the same method on the same object redundantly [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. ![MAJOR][MAJOR] [BlockChainImpl.java#L311](https://github.com/rsksmart/rskj/blob/892c99e1f7fc398495823679378f0f9f0bc504db/rskj-core/src/main/java/co/rsk/core/bc/BlockChainImpl.java#L311): Method co.rsk.core.bc.BlockChainImpl.internalTryToConnect(Block) appears to call the same method on the same object redundantly [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**